cmd/topicctl/subcmd/rebalance.go

```diff
@@ -0,0 +1,318 @@
+	dryRun                       bool
+	partitionBatchSizeOverride   int
+	pathPrefix                   string
+	autoContinueRebalance        bool
+	retentionDropStepDurationStr string
+	skipConfirm                  bool
+	sleepLoopDuration            time.Duration
+
+	shared sharedOptions
+
+	retentionDropStepDuration time.Duration
+}
+
+var rebalanceConfig rebalanceCmdConfig
+
+func init() {
+	rebalanceCmd.Flags().IntSliceVar(
+		&rebalanceConfig.brokersToRemove,
+		"to-remove",
+		[]int{},
+		"Brokers to remove; only applies if rebalance is also set",
+	)
+	rebalanceCmd.Flags().IntVar(
+		&rebalanceConfig.brokerThrottleMBsOverride,
+		"broker-throttle-mb",
+		0,
+		"Broker throttle override (MB/sec)",
+	)
+	rebalanceCmd.Flags().BoolVar(
+		&rebalanceConfig.dryRun,
+		"dry-run",
+		false,
+		"Do a dry-run",
+	)
+	rebalanceCmd.Flags().IntVar(
+		&rebalanceConfig.partitionBatchSizeOverride,
+		"partition-batch-size",
+		0,
+		"Partition batch size override",
+	)
+	rebalanceCmd.Flags().StringVar(
+		&rebalanceConfig.pathPrefix,
+		"path-prefix",
+		os.Getenv("TOPICCTL_APPLY_PATH_PREFIX"),
+		"Prefix for topic config paths",
+	)
+	rebalanceCmd.Flags().BoolVar(
+		&rebalanceConfig.autoContinueRebalance,
+		"auto-continue-rebalance",
+		false,
+		"Continue rebalancing without prompting (WARNING: user discretion advised)",
+	)
+	rebalanceCmd.Flags().StringVar(
+		&rebalanceConfig.retentionDropStepDurationStr,
+		"retention-drop-step-duration",
+		"",
+		"Amount of time to use for retention drop steps",
+	)
+	rebalanceCmd.Flags().BoolVar(
+		&rebalanceConfig.skipConfirm,
+		"skip-confirm",
+		false,
+		"Skip confirmation prompts during apply process",
+	)
+	rebalanceCmd.Flags().DurationVar(
+		&rebalanceConfig.sleepLoopDuration,
+		"sleep-loop-duration",
+		10*time.Second,
+		"Amount of time to wait between partition checks",
+	)
+
+	addSharedConfigOnlyFlags(rebalanceCmd, &rebalanceConfig.shared)
+	RootCmd.AddCommand(rebalanceCmd)
+}
+
+func rebalancePreRun(cmd *cobra.Command, args []string) error {
+	if rebalanceConfig.retentionDropStepDurationStr != "" {
+		var err error
+		rebalanceConfig.retentionDropStepDuration, err = time.ParseDuration(
+			rebalanceConfig.retentionDropStepDurationStr,
+		)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	// topicctl config layout for a kafka cluster (refer folder: topicctl/examples)
+	//   cluster/
+	//   cluster/cluster.yaml
+	//   cluster/topics/
+	//   cluster/topics/topic_1
+	//   cluster/topics/topic_2
+	//
+	// there can be a situation where topics folder is not in cluster dir
+	// hence we specify both --cluster-config and --path-prefix for action:rebalance
+	// However, we check the path-prefix topic configs with cluster yaml for consistency before applying
+	if rebalanceConfig.shared.clusterConfig == "" || rebalanceConfig.pathPrefix == "" {
+		return fmt.Errorf("Must set args --cluster-config & --path-prefix (or) env variables TOPICCTL_CLUSTER_CONFIG & TOPICCTL_APPLY_PATH_PREFIX")
```

Comment:
This wording is confusing. Can you clarify why does a `rebalance` command require both flags when `apply --rebalance` does not?